### PR TITLE
Add `Image` trait.

### LINKED ISSUE #367: [Feature request]: Add an `Image` trait as below
This is a proposal/feature request/RFC: Add the following trait to piet

```rust
pub trait Image {
    fn size(&self) -> Size;
}
```

It would be nice to to have to keep an `ImageBuf` around to know the size of the image.

### PR DESCRIPTION
**Pull request by derekdreery**
Closes #367.

The trait allows you to get an image's size in a cross-platform way.

Also renames `piet_common::Image` to `PietImage` to avoid clash. This is a breaking change, but if we want to add the `Image` trait and make the cross-platform type alias names more consistent, better to do it now than later.

### CODE REVIEW

**Comment by derekdreery** [comment]
I'd like to add a test, but it would be a merge conflict with #369, so I'd rather add it after that's merged.

**Comment by derekdreery** [comment]
Ready for review.

File: piet-cairo/src/lib.rs [hunk 1]
```diff
@@ -309,7 +298,13 @@ impl<'a> RenderContext for CairoRenderContext<'a> {
         dst_rect: impl Into<Rect>,
         interp: InterpolationMode,
     ) {
-        draw_image(self, image, Some(src_rect.into()), dst_rect.into(), interp);
+        draw_image(
```
Comment (cmyr) [inline piet-cairo/src/lib.rs:301]: it's strange to me that this isn't just a method on `CairoImageContext` itself, but that can be out of scope for this PR.

File: piet-common/tests/image.rs [hunk 2]
```diff
@@ -66,3 +66,19 @@ fn empty_image_area_should_not_panic() {
...
+        if image.size() != Size::new(1., 1.) {
+            return Err(format!(
+                "expected {:?}, found {:?}",
+                Size::new(1., 1.),
+                image.size()
+            ));
+        }
+        Ok(())
+    })
+}
```
Comment (cmyr) [inline piet-common/tests/image.rs:84]: do we also want to explicitly test a zero-sized image? I'm not sure what our expectations are there, now...

File: piet/src/image.rs [hunk 3]
```diff
@@ -22,6 +22,12 @@ use crate::kurbo::Size;
 use crate::util::unpremul;
 use crate::{Color, ImageFormat, RenderContext};
 
+/// Represents an image type that knows its size.
```
Comment (cmyr) [inline piet/src/image.rs:25]:
```suggestion
/// A trait for a backend's bitmap image type.
```

Don't love my version either, but in general I'd try and make the docs slightly higher level.

**Review by cmyr — APPROVED** [review]
I think this makes lots of sense, and generally makes the API more consistent.  Thanks!

File: piet-cairo/src/lib.rs (hunk 1 above)
Comment (derekdreery) [reply piet-cairo/src/lib.rs:301]: I've moved it to be a method on `CairoRenderContext`.

File: piet-common/tests/image.rs (hunk 2 above)
Comment (derekdreery) [reply piet-common/tests/image.rs:84]: Yeah, good idea.

**Comment by derekdreery** [comment]
I remember now, I actually solved the empty image issue for D2D by creating a transparent 1x1 pixel, which of course returns a size of `1x1`. I think the best solution is to add a flag to `Bitmap` to denote whether it's an empty image. I can then use this to completely skip rendering in the empty image case as well.
